u8timeseries/tests/test_autoregression_models.py

```diff
@@ -3,31 +3,67 @@
 import numpy as np
 
 from ..timeseries import TimeSeries
+from ..utils import timeseries_generation as tg
+from ..metrics import mape, overall_percentage_error, mase
 from u8timeseries import Prophet, KthValueAgoBaseline, ExponentialSmoothing, TimeSeries, Arima, AutoArima
 from u8timeseries.models.theta import Theta
 
-class ModelsTestCase(unittest.TestCase):
+class AutoregressionModelsTestCase(unittest.TestCase):
 
-    # forecasting horizon used in predictions
+    # forecasting horizon used in runnability tests
     forecasting_horizon = 5
 
-    # dummy timeseries for autoregression forecasting
-    times = pd.date_range('20000101', '20000130')
-    values = np.sin(range(len(times))) + np.array([2.0] * len(times))
-    ts: TimeSeries = TimeSeries.from_times_and_values(times, values)
+    # dummy timeseries for runnability tests
+    np.random.seed(0)
+    ts_gaussian = tg.gaussian_timeseries(length=100, mean=50)
 
+    # real timeseries for functionality tests
+    df = pd.read_csv('examples/AirPassengers.csv', delimiter=",")
+    ts_passengers = TimeSeries.from_dataframe(df, 'Month', '#Passengers')
+    ts_pass_train, ts_pass_val = ts_passengers.split_after(pd.Timestamp('19570101'))
 
-    def test_autoregressive_models_runnability(self):
-        models = [
-            ExponentialSmoothing(), 
-            Prophet(),
-            Arima(1, 1, 1),
-            AutoArima(),
-            KthValueAgoBaseline(),
-            Theta()
-        ]
+    # default autoregressive models
+    models = [
+        ExponentialSmoothing(), 
+        Prophet(),
+        Arima(1, 1, 1),
+        AutoArima(),
+        Theta()
+    ]
+    baseline_models = [
+        KthValueAgoBaseline()
+    ]
 
-        for model in models:
-            model.fit(self.ts)
+    # maximum error values for baselines
+    max_mape_baseline = 40
+
```

Comment:
It is quite cool to test both the models and the baselines. However here you introduce some coupling between them, which may render failures a bit less intuitive to interpret (did the baseline get better or did the model get worse?)

I think you could simply consider baselines as any other model, and test a list of models. Since this unit test is on a fixed dataset, we already know what MAPE to expect of each model, so we could use this as an upper bound (with some small tolerance). This way, you can have per-model thresholds, and test "more tightly" the models performing better. It would ensure that there is no performance regression if we change the models. For examples, if ExponentialSmoothing goes from 10% MAPE to 20% MAPE, we would catch it even if stays better than the baseline. You can also try building a few different versions of each model, with different parameters.

Reply:
done